iroha_p2p/src/message.rs

```diff
@@ -0,0 +1,30 @@
+use parity_scale_codec::{Decode, Encode};
+use std::io;
+
+#[derive(Debug, Clone, Encode, Decode, iroha_actor::Message)]
+pub enum Message {
+    Error,
+    Data { data: Vec<u8> },
+}
```

Comment:
I haven't seen error variant anywhere. Also if you are planning to use it from torii, then you can implement it in some fashion like this:
```rust
struct Message<T>(T)

impl<T: Encode> SimpleHandler<Message<T>> for Peer {
    fn handle(&mut self, msg: Message<T>) {
        self.send(msg.encode())
    }
}
```

That way you will make api of lib nicer and hide the encoding on network side

Reply:
Didn't quite understand what are do you mean by `self.send` in the code. Is it sending by network?

Reply:
yeah, send by network.

Reply:
I've simplified `Message` to a `struct Message(pub Vec<u8>)`, as this is what is read from stream.
Network part now doesn't encode/decode structs.
And sending data to peers is handled in a form of `SendPeerMsg` that already contains vector of bytes.

Reply:
I mean public api (for Network) should have handlers which generalize over Decode and Encode, hiding encoding/decoding inside library